crates/types/src/state_diffs.rs

```diff
@@ -59,3 +60,73 @@ where
         None => serializer.serialize_none(),
     }
 }
+
+impl ContractUpdate {
+    // Helper function to create a key for sorting
+    fn sort_key(&self) -> BigUint {
+        self.address.clone()
+    }
+
+    fn has_same_storage_updates(&self, other: &ContractUpdate) -> bool {
+        let mut self_storage = self.storage_updates.clone();
+        let mut other_storage = other.storage_updates.clone();
+
+        // Sort the storage updates by the unique key
+        self_storage.sort_by_key(|update| update.sort_key_storage());
+        other_storage.sort_by_key(|update| update.sort_key_storage());
+
+        if self_storage.len() != other_storage.len() {
+            return false;
+        }
+
+        self_storage
+            .iter()
+            .zip(other_storage.iter())
+            .all(|(self_update, other_update)| self_update == other_update)
+    }
+}
+
+impl StorageUpdate {
+    fn sort_key_storage(&self) -> BigUint {
+        self.key.clone()
+    }
+}
+impl PartialEq for ContractUpdate {
+    fn eq(&self, other: &Self) -> bool {
+        self.address == other.address
+            && self.nonce == other.nonce
+            && self.number_of_storage_updates == other.number_of_storage_updates
+            && self.new_class_hash == other.new_class_hash
+            && self.has_same_storage_updates(other)
+    }
+}
+
+impl DataJson {
+    pub fn has_same_contract_updates(&self, other: &DataJson) -> bool {
```

Comment:
I think we can have a function take generics as shown here - https://stackoverflow.com/questions/42748277/how-do-i-test-for-the-equality-of-two-unordered-lists

and use this for contract updates and class declarations. wdyt?

Reply:
this is good idea, let me do that